keyboards/keebio/quefrency/keymaps/marianas/relativity.c

```diff
@@ -0,0 +1,510 @@
+      register_code(KC_BSPC);
+      unregister_code(KC_BSPC);
+    }
+  }
+  printString(repl+i);
+}
+
+void deletePrev(void)
+{
+  if (countPointer == 0 && charCount[countPointer] == 0)
+    return;
+  for (int i = 0; i < charCount[countPointer]; i++)
+  {
+      register_code(KC_BSPC);
+      unregister_code(KC_BSPC);
+  }
+  charCount[countPointer] = 0;
+  int i = 1;
+  for (;i < tableNameListLen-1; i++)
+  {
+    if (tableNameList[i] == 0x0)
+    {
+      break;
+    }
+  }
+  tableNameList[i-1] = 0x0;
+  if (countPointer > 0)
+  {
+    countPointer--;
+  }
+}
+
+bool processSmartMacroTap(uint16_t kc)
+{
+  if (relativityTimer > 0 && TIMER_DIFF_16(timer_read(), relativityTimer) >= relTimeout)
+  {
+    deactivateRelativity();
+    return true;
+  }
+  relativityTimer = 0;
+  switch(kc)
+  {
+    case KC_C:
+      if (containsCode(KC_D))
+      {
+        printString("ribution");
+        printStringAndQueueChar("Center");
+      }
+      else if (last2CodeAre(KC_C))
+      {
+        ReplaceString("Corporation", "Contact");
+      }
+      else if(lastCodeIs(KC_C))
+      {
+        printString("oration");
+      }
+      else
+      {
+        printStringAndQueueChar("Corp");
+      }
+      break;
+    case KC_D:
+      if (last2CodeAre(KC_D))
+      {
+        ReplaceString("Distribution", "Distributor");
+      }
+      else if(lastCodeIs(KC_D))
+      {
+        printString("ribution");
+      }
+      else
+      {
+        printStringAndQueueChar("Dist");
+      }
+      break;
+    case KC_G:
+        printStringAndQueueChar("Global");
+        printStringAndQueueChar("Lookup");
+      break;
+    case KC_I:
+      if (containsCode(KC_W))
+        printStringAndQueueChar("Instance");
+      else
+        printStringAndQueueChar("Item");
+      break;
+    case KC_N:
+      printStringAndQueueChar("NadRate");
+      break;
+    case KC_P:
+      if (last2CodesAre(KC_D, KC_C))
+      {
+        ReplaceString("DistributionCenter", "DistCenter");
+        printStringAndQueueChar("Pricing");
+      }
+      else if (last2CodeAre(KC_P))
+      {
+      }
+      else if(lastCodeIs(KC_P))
+      {
+        ReplaceString("Product", "Person");
+      }
+      else
+      {
+        printStringAndQueueChar("Product");
+      }
+      break;
+    case KC_Q:
+      printStringAndQueueChar("Darden");
+      break;
+    case KC_S:
+      if (containsCode(KC_W))
+        if (containsCode(KC_S) || containsCode(KC_D))
+          printStringAndQueueChar("Step");
+        else
+          printStringAndQueueChar("Session");
+      else
+        printStringAndQueueChar("Supplier");
+      break;
+    case KC_T:
+      if (containsCode(KC_W))
+        printStringAndQueueChar("Task");
+      else
+        printStringAndQueueChar("Type");
+      break;
+    case KC_W:
+      printStringAndQueueChar("Workflow");
+      break;
+  }
+  addKeyCode(kc);
+  return false;
+}
+
+
+bool shifted = false;
+bool isShifted()
+{
+  return shifted;
```

Comment:
This may be a better way to go this. 

Real Mods is when you hold a the shift key, space cadet, or hold oneshot mods down.  Weak mods are used elsewhere for stuff like macros or mod tap.  
```suggestion
  return  ( get_mods() | get_weak_mods() ) & MOD_MASK_SHIFT;
```

So, this may be more accurate without the manual tracking.